index.js

```diff
@@ -15,164 +23,341 @@ const supportedHelpers = [
   'TestCafe',
   'Playwright'
 ];
-let launchObj;
-let itemObj;
-let fileName;
-let stepInfo;
-let rpClient;
-let logFile;
-let suiteTempId;
-let beforeSuiteStatus = 'failed';
-let launchStatus = 'passed';
-
-class ReportPortalHelper extends Helper {
-  _updateStep(step, status) {
-    this._finishTestItem(launchObj, itemObj, step, status);
-  }
 
-  _passed() {
-    this._updateStep(stepInfo, 'passed');
+for (const helperName of supportedHelpers) {
+  if (Object.keys(helpers).indexOf(helperName) > -1) {
+    helper = helpers[helperName];
   }
+}
 
-  async _failed(test) {
-    launchStatus = 'failed';
-    this.errMsg = test.err.message;
-    this._updateStep(stepInfo, 'failed');
-  }
+const defaultConfig = {
+  token: '',
+  endpoint: '',
+  project: '',
+  launchDescription: '',
+  attributes: [],
+  debug: false,
+  rerun: undefined,
+  enabled: false
+};
 
-  _startLaunch(suiteTitle) {
-    rpClient = new RPClient({
-      token: this.config.token,
-      endpoint: this.config.endpoint,
-      project: this.config.projectName,
-      debug: this.config.debug,
-    });
+module.exports = (config) => {
+  config = Object.assign(defaultConfig, config);
 
-    return rpClient.startLaunch({
-      name: this.config.launchName || suiteTitle,
-      description: this.config.launchDescription || '',
-      attributes: this.config.launchAttributes || [],
-      rerun: this.config.rerun || false,
-      rerunOf: this.config.rerunOf || undefined,
+  console.log('Staaarting!!!11');
+
+  let launchObj;
+  let suiteObj;
+  let testObj;
+  let fileName;
+  let stepInfo;
+  let stepObj;
+  let rpClient;
+  let logFile;
+  let suiteStatus = 'PASSED';
+  let launchStatus = 'PASSED';
+  let currentMetaSteps = [];
+
+
+  event.dispatcher.on(event.all.before, () => {
+    launchObj = startLaunch();
+    console.log(`${launchObj.tempId}: The launchId is started.`);
```

Comment:
maybe we can try to use the `debug: true` of RP, then it will print out all those things 😸

Reply:
Yeah, was thinking about it as well
Also about using `debug` function
Once I make it work I will clean it up